packages/jest-util/src/create_process_object.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import deepCyclicCopy from './deep_cyclic_copy';
+
+export default function() {
+  const newProcess = deepCyclicCopy(process);
+
+  // $FlowFixMe: Add the symbol for toString objects.
+  newProcess[Symbol.toStringTag] = 'process';
+
+  // Sequentially execute all constructors over the object.
+  for (let chain = process; chain; chain = Object.getPrototypeOf(chain)) {
```

Comment:
Woah what the hell is this for loop. What about do-while?

Reply:
You don't have to change it, it's just very confusing to see "chain;" in there, took me a second :D

Reply:
Let me change it; I don't want confusing code and especially in that file :)